File: resources/js/components/FormSelect.js
import React from 'react'
import PropTypes from 'prop-types'

class FormSelect extends React.Component {
  render() {
    let options = this.props.options.map(option => {
      return (
        <option key={option.value} value={option.value}>
          {option.text}
        </option>
      )
    })
    options.unshift(
      <option key="" value="">
        {/* {this.props.placeholder} */}
        Open this select menu
      </option>
    )
    return (
      <div className="mb-3">
        <label htmlFor={this.props.name}>{this.props.placeholder}</label>
        <select
          className={
            this.props.field.error ? 'form-select is-invalid' : 'form-select'
          }
          name={this.props.name}
          id={this.props.name}
          value={this.props.field.value}
          onChange={this.props.onChange}
          aria-label={this.props.placeholder}
        >
          {options}
        </select>
        {this.props.field.error && (
          <label className="invalid-feedback" htmlFor={this.props.name}>
            {this.props.field.error}
          </label>
        )}
      </div>
    )
  }
}

FormSelect.propTypes = {
  field: PropTypes.shape({
    error: PropTypes.string,
    value: PropTypes.string,
  }),
  name: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string,
      text: PropTypes.string,
    })
  ),
}

export default FormSelect

File: resources/js/components/FormSelect.fs
module FormSelect

open Feliz
open Main

let selectPlaceholder = Html.option [
    prop.value ""
    prop.key ""
    prop.text "Open this select menu"
]

[<ReactComponent>]
let FormSelect(name: string,
               placeholder: string, 
               options: SelectOption array, 
               field: Field, 
               onChange: Browser.Types.Event -> unit) : ReactElement =
    Html.div [
        prop.className "mb-3"
        prop.children [
            Html.label [
                prop.htmlFor name
                prop.text placeholder
            ] 
            Html.select [
                prop.className 
                    (if String.length field.error > 0 
                     then "form-control is-invalid" 
                     else "form-control") 
                prop.name name
                prop.id name
                prop.value field.value
                prop.onChange onChange
                prop.ariaLabel placeholder
                prop.children (Array.append 
                    [|selectPlaceholder|] 
                    (Array.map (fun x -> 
                        Html.option [
                            prop.value x.value
                            prop.key x.value
                            prop.text x.text
                        ]) options))
            ]
            if String.length field.error > 0 then Html.label [
                prop.className "invalid-feedback"
                prop.htmlFor name
                prop.text field.error
            ]
        ]
    ]